tests/models/layoutlmv3/test_modeling_tf_layoutlmv3.py

```diff
@@ -0,0 +1,444 @@
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        (config, input_ids, bbox, pixel_values, token_type_ids, input_mask, _, _) = config_and_inputs
+        inputs_dict = {
+            "input_ids": input_ids,
+            "bbox": bbox,
+            "pixel_values": pixel_values,
+            "token_type_ids": token_type_ids,
+            "attention_mask": input_mask,
+        }
+        return config, inputs_dict
+
+
+@require_tf
+class TFLayoutLMv3ModelTest(TFModelTesterMixin, unittest.TestCase):
+
+    all_model_classes = (
+        (
+            TFLayoutLMv3Model,
+            TFLayoutLMv3ForQuestionAnswering,
+            TFLayoutLMv3ForSequenceClassification,
+            TFLayoutLMv3ForTokenClassification,
+        )
+        if is_tf_available()
+        else ()
+    )
+
+    test_pruning = False
+    test_resize_embeddings = False
+    test_onnx = False
+
+    def _prepare_for_class(self, inputs_dict, model_class, return_labels=False) -> dict:
+        inputs_dict = copy.deepcopy(inputs_dict)
+
+        if model_class in get_values(TF_MODEL_FOR_MULTIPLE_CHOICE_MAPPING):
+            inputs_dict = {
+                k: tf.tile(tf.expand_dims(v, 1), (1, self.model_tester.num_choices) + (1,) * (v.ndim - 1))
+                if isinstance(v, tf.Tensor) and v.ndim > 0
+                else v
+                for k, v in inputs_dict.items()
+            }
+
+        if return_labels:
+            if model_class in get_values(TF_MODEL_FOR_MULTIPLE_CHOICE_MAPPING):
+                inputs_dict["labels"] = tf.ones(self.model_tester.batch_size, dtype=tf.int32)
+            elif model_class in get_values(TF_MODEL_FOR_QUESTION_ANSWERING_MAPPING):
+                inputs_dict["start_positions"] = tf.zeros(self.model_tester.batch_size, dtype=tf.int32)
+                inputs_dict["end_positions"] = tf.zeros(self.model_tester.batch_size, dtype=tf.int32)
+            elif model_class in get_values(TF_MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING):
+                inputs_dict["labels"] = tf.zeros(self.model_tester.batch_size, dtype=tf.int32)
+            elif model_class in get_values(TF_MODEL_FOR_TOKEN_CLASSIFICATION_MAPPING):
+                inputs_dict["labels"] = tf.zeros(
+                    (self.model_tester.batch_size, self.model_tester.text_seq_length), dtype=tf.int32
+                )
+
+        return inputs_dict
+
+    def setUp(self):
+        self.model_tester = TFLayoutLMv3ModelTester(self)
+        self.config_tester = ConfigTester(self, config_class=LayoutLMv3Config, hidden_size=37)
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    @unittest.skip(
+        reason="""
+    We disable this test from TFModelTesterMixin because it assumes
+    that the model only accepts input_ids or pixel_values, but not both.
+    That assumption sometimes causes the test to break, although the model works fine.
```

Comment:
This is odd. I'm not sure why it would only sometimes break with this assumption and we have other models that accept both `pixel_values` and `input_ids`. What is the error you find?

Reply:
Get an error of incompatible shapes because pixel_values sometimes becomes the first argument to the model, which means that input_ids get set as pixel_values. It then propagates down to TFLayoutLMv3TextEmbeddings where pixel_values is used as argument to the word_embeddings and the error occurs in line 265 of modeling_tf_layoutlmv3.py since it tries to add input_embeds (which has the shape from pixel_values) and token_type_embeddings (which havs the shape from the sequence length).

The reason that it only happens sometimes is due to these lines in test_loss_computation:

``` py
possible_input_names = {"input_ids", "pixel_values", "input_features"}
input_name = possible_input_names.intersection(set(prepared_for_class)).pop()
model_input = prepared_for_class.pop(input_name)
```

input_ids and pixel_values are both in prepared_for_class, so both will occur in the set intersection. Since sets don't have an order, then we sometimes get input_ids as the model_input and sometimes we get pixel_values as the model_input.

This makes the model call fail when model_input is set to pixel_values:

``` py
loss = model(model_input, **prepared_for_class)[0]
```

I am sure I can customise the test to work with this model, but I am a bit curious as to why it hasn't failed for other models that accepts both inputs.